src/action.js

```diff
@@ -31,34 +31,49 @@ module.exports = function link(config, args) {
   }
 
   const packageName = args[0];
-  const dependencies = packageName ? [packageName] : getProjectDependencies();
 
-  dependencies
-    .forEach(name => {
-      const dependencyConfig = config.getDependencyConfig(name);
-
-      if (!dependencyConfig) {
-        return log.warn('ERRINVALIDPROJ', `Project ${name} is not a react-native library`);
+  const dependencies = (packageName ? [packageName] : getProjectDependencies())
+    .map(name => ({
+      config: config.getDependencyConfig(name),
+      name,
+    }))
+    .filter(dependency => {
+      if (!dependency.config) {
+        log.warn('ERRINVALIDPROJ', `Project ${dependency.name} is not a react-native library`);
+        return false;
       }
+      return true;
+    });
 
-      if (project.android && dependencyConfig.android) {
+  dependencies
+    .forEach(dependency => {
+      if (project.android && dependency.config.android) {
         log.info(`Linking ${name} android dependency`);
-        registerDependencyAndroid(name, dependencyConfig.android, project.android);
+        registerDependencyAndroid(name, dependency.android, project.android);
       }
 
-      if (project.ios && dependencyConfig.ios) {
+      if (project.ios && dependency.config.ios) {
         log.info(`Linking ${name} ios dependency`);
-        registerDependencyIOS(dependencyConfig.ios, project.ios);
+        registerDependencyIOS(dependency.config.ios, project.ios);
       }
+    });
 
-      if (project.android && !isEmpty(dependencyConfig.assets)) {
-        log.info(`Copying assets from ${name} to android project`);
-        copyAssetsAndroid(dependencyConfig.assets, project.android.assetsPath);
-      }
+  const assets = dependencies.reduce(
+    (assets, dependency) => assets.concat(dependency.config.assets),
+    project.assets
+  );
```

Comment:
We have to map dependencies first so we run copyAssets at once for all dependencies & project so that we can dedupe the array by basename quite easily and avoid conflicts (e.g. duplicated font names). That way it's imo clearer.